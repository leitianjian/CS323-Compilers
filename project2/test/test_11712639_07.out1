Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: t
        LP
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            RETURN
            Exp (2)
              INT: 0
            SEMI
        RC
    ExtDefList (4)
      ExtDef (4)
        Specifier (4)
          TYPE: int
        FunDec (4)
          ID: test
          LP
          VarList (4)
            ParamDec (4)
              Specifier (4)
                TYPE: int
              VarDec (4)
                ID: n
            COMMA
            VarList (4)
              ParamDec (4)
                Specifier (4)
                  TYPE: int
                VarDec (4)
                  ID: m
          RP
        CompSt (4)
          LC
          StmtList (5)
            Stmt (5)
              Exp (5)
                Exp (5)
                  CHAR: 'c'
                ASSIGN
                Exp (5)
                  ID: n
              SEMI
            StmtList (6)
              Stmt (6)
                Exp (6)
                  Exp (6)
                    ID: n
                  ASSIGN
                  Exp (6)
                    Exp (6)
                      INT: 1
                    ASSIGN
                    Exp (6)
                      INT: 2
                SEMI
              StmtList (7)
                Stmt (7)
                  Exp (7)
                    Exp (7)
                      Exp (7)
                        ID: n
                      EQ
                      Exp (7)
                        ID: m
                    ASSIGN
                    Exp (7)
                      INT: 2
                  SEMI
                StmtList (8)
                  Stmt (8)
                    Exp (8)
                      Exp (8)
                        ID: n
                      ASSIGN
                      Exp (8)
                        Exp (8)
                          ID: m
                        ASSIGN
                        Exp (8)
                          INT: 2
                    SEMI
                  StmtList (9)
                    Stmt (9)
                      Exp (9)
                        Exp (9)
                          ID: t
                        ASSIGN
                        Exp (9)
                          ID: n
                      SEMI
                    StmtList (10)
                      Stmt (10)
                        Exp (10)
                          Exp (10)
                            LP
                            Exp (10)
                              ID: n
                            RP
                          ASSIGN
                          Exp (10)
                            ID: m
                        SEMI
                      StmtList (11)
                        Stmt (11)
                          Exp (11)
                            Exp (11)
                              ID: t
                              LP
                              RP
                            ASSIGN
                            Exp (11)
                              CHAR: 'c'
                          SEMI
                        StmtList (12)
                          Stmt (12)
                            Exp (12)
                              Exp (12)
                                MINUS
                                Exp (12)
                                  ID: n
                              ASSIGN
                              Exp (12)
                                ID: m
                            SEMI
                          StmtList (13)
                            Stmt (13)
                              Exp (13)
                                Exp (13)
                                  Exp (13)
                                    ID: n
                                  AND
                                  Exp (13)
                                    ID: m
                                ASSIGN
                                Exp (13)
                                  INT: 1
                              SEMI
                            StmtList (14)
                              Stmt (14)
                                Exp (14)
                                  Exp (14)
                                    ID: n
                                  ASSIGN
                                  Exp (14)
                                    Exp (14)
                                      ID: m
                                    AND
                                    Exp (14)
                                      INT: 1
                                SEMI
                              StmtList (15)
                                Stmt (15)
                                  Exp (15)
                                    Exp (15)
                                      Exp (15)
                                        ID: n
                                      LT
                                      Exp (15)
                                        ID: m
                                    ASSIGN
                                    Exp (15)
                                      INT: 2
                                  SEMI
                                StmtList (16)
                                  Stmt (16)
                                    Exp (16)
                                      Exp (16)
                                        Exp (16)
                                          ID: n
                                        PLUS
                                        Exp (16)
                                          ID: m
                                      ASSIGN
                                      Exp (16)
                                        INT: 3
                                    SEMI
          RC
