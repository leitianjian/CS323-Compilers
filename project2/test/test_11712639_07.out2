Error type 6 at Line 5: rvalue on the left side of assignment operator
Error type 6 at Line 6: rvalue on the left side of assignment operator
Error type 5 at Line 6: unmatching type on both sides of assignment
Error type 6 at Line 7: rvalue on the left side of assignment operator
Error type 5 at Line 9: unmatching type on both sides of assignment
Error type 6 at Line 11: rvalue on the left side of assignment operator
Error type 6 at Line 12: rvalue on the left side of assignment operator
Error type 6 at Line 13: rvalue on the left side of assignment operator
Error type 6 at Line 15: rvalue on the left side of assignment operator
Error type 6 at Line 16: rvalue on the left side of assignment operator
