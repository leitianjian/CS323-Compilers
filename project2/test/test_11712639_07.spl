int t(){
    return 0;
}
int test(int n, int m){
    'c' = n;
    n = 1 = 2;
    n == m = 2;
    n = m = 2;
    t = n;
    (n) = m;
    t() = 'c';
    -n = m;
    n && m = 1;
    n = m && 1;
    n < m = 2;
    n + m = 3;
}