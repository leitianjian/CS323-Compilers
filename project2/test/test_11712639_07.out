Error type 6 at line 5: rvalue on the left side of assignment operator
Error type 5 at line 5: unmatching types on both sides of assignment operator
Error type 6 at line 6: rvalue on the left side of assignment operator
Error type 6 at line 7: rvalue on the left side of assignment operator
Error type 6 at line 9: rvalue on the left side of assignment operator
Error type 6 at line 10: rvalue on the left side of assignment operator
Error type 6 at line 11: rvalue on the left side of assignment operator
Error type 5 at line 11: unmatching types on both sides of assignment operator
Error type 6 at line 12: rvalue on the left side of assignment operator
Error type 6 at line 13: rvalue on the left side of assignment operator
Error type 6 at line 15: rvalue on the left side of assignment operator
Error type 6 at line 16: rvalue on the left side of assignment operator
